[IV-23-24] Objetivo 3: Automatización de operaciones

marinajcs opened the pull request:
* [x] ¿Incluye el fichero `iv.yaml` las claves de todos los objetivos
anteriores?
* [x] ¿Se ha desarrollado el fichero para las tareas a partir de un
   issue?
* [x] ¿Ese issue está asignado a un milestone?
* [x] ¿El milestone incluye, entre sus productos mínimamente viables, la
  existencia de herramientas que permitan automatizar tareas?
* [x] ¿Hemos justificado correctamente en caso de elegir un gestor de
   dependencias el que se use como gestor de tareas?
* [x] ¿El PR, que también es un issue, está asignado al mismo milestone?
* [x] ¿La tarea `check` efectivamente comprueba la sintaxis de los fuentes
  existentes y sólo eso?
* [x] ¿Se ha comprobado que las instrucciones incluidas en el `README.md` son
completas y correctas?

marinajcs committed:
Añadido fichero de configuración tsconfig.ts

Servirá para definir las configruaciones del compilador de TypeScript para el proyecto en concreto

marinajcs committed:
Elección del runtime, closes #20

Justificación de la decisión de Niode.js en el fichero añadido

marinajcs committed:
Elección del gestor de dependencias, closes #18

Justificación de la decisión de pnpm en el fichero añadido

marinajcs committed:
package.json y lock generados con pnpm #18

Ficheros generados automáticamente tras la ejecución de pnpm init y pnpm install, gestionan la configuración de proyectos Node.js y las dependencias

marinajcs committed:
Elección del gestor de tareas closes #19

Justificación de la decisión Grunt en el fichero correspondiente

marinajcs committed:
Añadida tarea de comprobación de sintaxis closes #22

Es conveniente poder verificar la sintaxis del código hasta ahora incluido

marinajcs committed:
Ligeras modificaciones en gestor-dependencias.md

PabloBarTo reviewed (approved):
Yo lo veo en general todo bien. Aunque por poner alguna pega, a la hora de elegir cualquier gestor definiría unos criterios antes de las explicaciones de los gestores y basaría las explicaciones de estos en esos criterios.

marinajcs committed:
Incluidos criterios de elección de runtime, fixes #20

Se han descrito previamente una serie de criterios que se han tomado en cuenta para la elección de un runtime adecuado para el proyecto

marinajcs committed:
Arreglado pequeño error sintáctico, fixes #20

marinajcs committed:
Incluidos criterios elección de gestor-dependencias, fixes #18

Se han redactado previamente los criterios a tomar en cuenta en la elección del gestor de dependencias para facilitar el proceso de selección.

marinajcs committed:
Incluidos criterios selección de gestor-tareas, fixes #19

Se han incluido previamente los criterios seguidos a la hora de decantarse por un gestor de tareas para mejorar el proceso de selección.

marinajcs commented:
> Yo lo veo en general todo bien. Aunque por poner alguna pega, a la hora de elegir cualquier gestor definiría unos criterios antes de las explicaciones de los gestores y basaría las explicaciones de estos en esos criterios.

Entendido, gracias por la sugerencia

marinajcs commented:
Listo para revisión @JJ

JJ commented on Gruntfile.js:
```diff
@@ -0,0 +1,17 @@
+module.exports = function (grunt) {
+    grunt.initConfig({
+        // Comprobación de sintaxis
+        eslint: {
```
eslint es un comprobador de buenas prácticas, no de sintaxis.

marinajcs replied:
Corregido en https://github.com/marinajcs/asignacionTareas/pull/21/commits/5bd71d42f4224c1aad2dae6a474a512ac6ec30ff

JJ reviewed (changes requested):
Antes de revisar con más extensión el resto, mira que en la lista de comprobación pone explícitamente que se debe comprobar sólo la sintaxis. Se trata de que no necesitéis instalar ninguna herramienta externa, precisamente.

marinajcs committed:
Eliminado .eslintrc.js, fixes #22

Es lint es un comprobador de buenas prácticas, no de sintaxis

marinajcs committed:
Modificada comprobación de sintaxis, fixes #22

Se cambia es lint, que es un verificador de buenas prácticas, por un comando de tsc que comprueba únicamente la sintaxis del código

JJ commented on docs/gestor-tareas.md:
```diff
@@ -0,0 +1,56 @@
+
+### gulp
+
+* *Descripción*: Gulp es un sistema de construcción y automatización de tareas para JavaScript. Se utiliza para realizar tareas repetitivas como la minificación de archivos, la compilación de preprocesadores y más.
```
Gulp está irremediablemente obsoleto, me temo.

marinajcs replied:
Corregido en https://github.com/marinajcs/asignacionTareas/pull/21/commits/06fb0dc880ac99d3d56d8be3c6c7c92e8d5258c1

JJ commented on docs/gestor-tareas.md:
```diff
@@ -0,0 +1,56 @@
+
+## Elección final
+
+Para el desarrollo de una solución para el problema de la asignación de tareas en un piso de estudiantes, tanto Gulp como Grunt pueden ser adecuados; sin embargo, podría tener sentido 
```
Podías haber incluido aquí también npm run, pnpm run y yarn run, que los vas a tener disponibles siempre.

marinajcs replied:
Corregido en https://github.com/marinajcs/asignacionTareas/pull/21/commits/06fb0dc880ac99d3d56d8be3c6c7c92e8d5258c1

JJ commented on docs/runtime.md:
```diff
@@ -0,0 +1,57 @@
+Creado por el mismo que desarrolló Node.js, fue diseñado para abordar ciertos
+problemas de seguridad y diseño.
+
+### Navegador
```
Esto no hace falta que lo pongas. No vas a ejecutar un servidor web en el navegador.

marinajcs replied:
Corregido en https://github.com/marinajcs/asignacionTareas/pull/21/commits/dd57e1dd7ad5311684205aa9370680d4614d50e7

JJ commented on package.json:
```diff
@@ -1,22 +1,21 @@
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^6.11.0",
+    "@typescript-eslint/parser": "^6.11.0",
+    "eslint": "^8.53.0",
```
No has quitado las dependencias que no vas a usar. Deberías desinstalarlas de la forma correcta.

marinajcs replied:
Corregido en https://github.com/marinajcs/asignacionTareas/pull/21/commits/9340bc8648701019d25510853b614bcbd63e7cbf

JJ reviewed (changes requested):
Realmente no hay ninguna de las objeciones que sea crítica; si acaso el no haber desinstalado eslint de la forma correcta, pero en conjunto sí es cierto que podría mejorar un poco para superar el objetivo. Está muy cerca, en todo caso, y gracias a @PabloBarTo por la ayuda prestada.

marinajcs committed:
Eliminadas dependencias de forma correcta

Se eliminan las dependencias no usadas mediante pnpm remove para borrarlas de forma limpia

marinajcs committed:
Descartada opción del navegador, fixes #20

No se va a ejecutar un servidor web en el navegador, por lo que se descarta directamente

marinajcs committed:
Modificadas las opciones consideradas, fixes #19

Se ha descartado por completo Gulp, que está obsoleto, y se han añadido las opciones de npm/pnpm/yarn run, que siempre estarán disponibles.

marinajcs committed:
Eliminada mención a gulp en gestor-tareas.md, fixes #19

JJ commented on docs/gestor-dependencias.md:
```diff
@@ -0,0 +1,72 @@
+
+## Opciones consideradas
+
+### deno
```
Deno sólo es una opción si lo eliges como runtime. Bun sí es una opción, porque es compatible con node. Por favor, trata de repasar qué hace cada uno.

marinajcs replied:
Corregido en https://github.com/marinajcs/asignacionTareas/pull/21/commits/d7f9ee6f38ce81cacfdc4ead2df6c1150cd5f5a6

Entendido, no tenía sentido incluir Deno entre las opciones de gestor de dependencias si no se ha elegido como runtime. También he incluido Bun como opción tanto en runtime como en gestor de dependencias, ya que sí es
compatible con node.

JJ reviewed (changes requested):
Me temo que deno no es un gestor de dependencias, salvo que elijas deno como runtime y efectivamente uses sus ficheros para gestionar dependencias. Por favor, repasa qué son deno, bun y node y ajusta las elecciones y documentación en consonancia.

marinajcs committed:
Modificadas las opciones consideradas, fixes #18

Se descarta Deno como gestor de dependencias al no haberlo escogido como runtime, y se incluye como pposible opción Bun, que es compatible con Node.js.

JJ reviewed (approved):
:+1:
